### PR TITLE
Add integration test for equipping/unequipping clothing entities

### LINKED ISSUE #38194: Need integration test for (un)equipping clothing items
We got quite a few clothing items that add actions, spawn entities, add overlays etc. when equipped.
So in order to catch errors we could use a simple test that

- spawns every item with the `ClothingComponent`
- spawns an Urist
- equips the clothing items into the corresponding inventory slot (some items might fit into multiple slots)
- simulates the game for a few ticks
- unequips the item
- simulates again for a few ticks

Note that some inventory slots depend on others, so that Urist might need to wear a jumpsuit so it can equip pdas or similar.
Also there are quite a lot of clothing items, so maybe the test could use some optimization to skip spawning those that do absolutely nothing.
And it is important to test this with a client, not only on the server, so that all overlays are being added.

### PR DESCRIPTION
**Pull request by perryprog**
## About the PR

This PR requires #38274, and will be marked as a draft until it's merged.

This introduces an integration test that, at the broad level:
1. Spawns in every item that has a ClothingComponent and then
2. Equips each one (plus jumpsuit/hardsuit if needed) into each slot it can fit in
3. Triggers any InstantActions it gains on each equip
4. Unequips it (and jumpsuit/hardsuit)

## Why / Balance

Fixes #38194. Would have caught part of #38274 (equip mask, toggle mask, unequip mask), as well as #38286.

## Technical details

- This is my first time writing an integration test for SS14—any nitpicks are welcome. I'll do a self review on a few bits I have questions about.
- Running the test on a downstream passed without issue.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    {
+        var clothingProtos = ProtoMan.EnumeratePrototypes<EntityPrototype>()
+            .Where(x => x.HasComponent<ClothingComponent>()
+                        && !x.HideSpawnMenu
+                        && !x.Abstract
+                        && !Pair.IsTestPrototype(x)
+                        && !x.HasComponent<RandomSpawnerComponent>()
+                        // Mobs are a massive pain in this test
+                        // (Admeme mouse eating your cak, cancer mouse killing you... mice just dying...)
+                        && !x.HasComponent<MobStateComponent>());
```
Comment (perryprog) [inline Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs]: Previously it was mentioned that it might be good to filter out clothing items that don't do anything, however, it turns out this test also serves as a way to test for the validity of sprites in each possible equip position. This makes me think it might be good to keep this broad-as is, but this is definitely on the slow side of things. It seems to take about 30–40 seconds to run for me, though I think part of that is due to maybe some unneeded RunTicks?

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        EntityUid hardsuit = default!;
+        await Server.WaitPost(() =>
+        {
+            foreach (var clothing in clothingProtos)
+            {
+                var ent = SEntMan.SpawnAtPosition(clothing.ID, ToServer(PlayerCoords));
+                clothings.Add(ent);
+
+                // Material bag can pick up other clothing items.
+                SEntMan.RemoveComponent<MagnetPickupComponent>(ent);
```
Comment (perryprog) [inline Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs]: Ew icky hardcoded edge cases. Should I move this into some sort of `removedComponents` variable? (And probably also an `ignoredComponents` one for use in the prototype filtering?)

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+                    DoEquip(invSystem, clothing, slot.Name, clothingComp);
+                });
+                await RunTicks(1);
+
+                var newActions = actionSys.GetActions(SPlayer)
+                    .Except(baseActions)
+                    // Don't bother with more complicated actions (they can take targets, etc.)
+                    .Where(action => SEntMan.HasComponent<InstantActionComponent>(action.Owner))
+                    .ToList();
```
Comment (perryprog) [inline Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs:121]: While this does end up passing on both here and a downstream I tried the test on, this feels like it's asking for trouble. Consider an item that grants an instant action which scrambles you. That would then put the player out of range of other items, killing the test.

Should this be filtered down to just actions that have a ToggleAction component? Or even just ToggleClothing?

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+                var unremovableComp = SEntMan.GetComponentOrNull<UnremoveableComponent>(clothing);
+                // Electropack or similar has to be force removed, since our unequip actor is always ourselves
+                var needsForce = SEntMan.HasComponent<SelfUnremovableClothingComponent>(clothing);
+
+                if (unremovableComp != null)
+                {
+                    needsForce = true;
+                    // We want to reuse this entity for other slots. (E.g., CluwnePDA)
+                    unremovableComp.DeleteOnDrop = false;
```
Comment (perryprog) [inline Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs:134]: Unrelated but there might be an odd transient bug in handling action serialization and DeleteOnDrop unremovable items—I was only able to reproduce it once, though, when force unequipping a cluwne PDA. Was kinda odd.

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+                    // If we should've needed to force for this item, but got it off, that's no good.
+                    Assert.That(isRemoved && needsForce,
+                        Is.False,
+                        $"Unequipped {SEntMan.ToPrettyString(clothing)}, which should have been unremovable.");
+
+                    // Just in case some kind of toggle action put us into a state that this test doesn't expect.
+                    // Currently this doesn't seem to get hit.
+                    if (!isRemoved)
+                        RunActions(newActions, actionSys);
```
Comment (perryprog) [inline Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs:151]: Should this be removed? In the original implementation I thought it was needed, but I'm not seeing any cases where this is a behavior that exists.

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+/// <summary>
+/// This test spawns in every item that has a ClothingComponent and then simulates:
+/// 1. Equipping it (and jumpsuit/hardsuit if needed)
+/// 2. Triggering any InstantActions it grants.
+/// 3. Unequipping it.
+/// </summary>
+public sealed class ClothingEquipTest : InteractionTest
+{
+    private const string HardsuitProto = "ClothingOuterHardsuitEVA";
```
Comment (Tayrtahn) [inline Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs]:
```suggestion
    private static readonly EntProtoId HardsuitProto = "ClothingOuterHardsuitEVA";
```
Make these static `EntProtoId`s so they get validated.

**Comment by perryprog** [comment]
Also—I don’t believe this covers most/all overlays, as it seems their draw code is only invoked in the full game loop during rendering. (I thiiiiink.) I’m not sure if there’s a way we can configure this test to run with that?

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs (hunk 4 above)
Comment (perryprog) [reply Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs:134]: Lol it was #38293.

File: Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs (hunk 6 above)
Comment (perryprog) [reply Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs]: Guh, I had that on my mental list to fix, too. Thanks!
